tests/test_pdf_polynomials.py

```diff
@@ -0,0 +1,28 @@
+#  Copyright (c) 2019 zfit
+
+
+import pytest
+import zfit
+from zfit import ztf
+
+from zfit.core.testing import setup_function, teardown_function, tester
+
+obs1 = zfit.Space(obs="obs1", limits=(-0.9, 0.85))
```

Comment:
This should be (-1, 1) to be canonical, shouldn't it?

Reply:
I agree, for the canonical case this should be (-1, 1) ...
Now have you changed also this to receive any other non-canonical range and re-scale back?

Reply:
Yes, it simply scales when calculating the polynomial, building them in the interval [a,b] as orthogonal.

Reply:
perfect! I will give a try once I get to the office

Reply:
We need to write proper tests for it. Also, you should now be able to use the polynomials as PDFs with coefficients.

Reply:
ok, but what I didn't understand is wether we have already an implementation that I can provide the parameters or whether I need to build the c0*Chebby(order1) + c1*Cheby(order2) + ...?

Reply:
No, now you do

```
Cheby(obs=obs, [c0, c1, c2, ...])
```

and the degree is going to be inferred from there.

Reply:
perfect!

Reply:
It's not (-1, 1) to test the analytic integral, since (-1, 1) is already normalized to 1.
But of course, both should be tested, the tests are yet incomplete.

On the scaling: We also need to rescale the analytic integration then (the limits given there), right?
I'll do that with some clean up

Reply:
True, the analytic integral should be rescaled.

Reply:
Given another try on this, I see the following issue:
 `Traceback (most recent call last):
  File "simpleBkgFit.py", line 72, in <module>
    minimum = minimizer.minimize(loss=nll)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 205, in minimize
    return self._hook_minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 210, in _hook_minimize
    return self._call_minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 214, in _call_minimize
    return self._minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/minimizer_minuit.py", line 48, in _minimize
    gradients = loss.gradients(params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 147, in gradients
    return self._gradients(params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 240, in _gradients
    gradients = {(p, grad) for p, grad in zip(params_todo, super()._gradients(params_todo))}
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 209, in _gradients
    return tf.gradients(self.value(), params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 187, in value
    return self._value()
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 223, in _value
    loss = super()._value()
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 192, in _value
    constraints=self.constraints)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 255, in _loss_func
    nll = _unbinned_nll_tf(model=model, data=data, fit_range=fit_range)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 38, in _unbinned_nll_tf
    for p, d, r in zip(model, data, fit_range)]
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 38, in <listcomp>
    for p, d, r in zip(model, data, fit_range)]
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/loss.py", line 42, in _unbinned_nll_tf
    probs = model.pdf(data, norm_range=fit_range)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 314, in pdf
    value = self._single_hook_pdf(x=x, norm_range=norm_range, name=name)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 324, in _single_hook_pdf
    return self._hook_pdf(x=x, norm_range=norm_range, name=name)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 327, in _hook_pdf
    return self._norm_pdf(x=x, norm_range=norm_range, name=name)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 330, in _norm_pdf
    return self._call_pdf(x=x, norm_range=norm_range, name=name)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 338, in _call_pdf
    return self._fallback_pdf(x=x, norm_range=norm_range)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 341, in _fallback_pdf
    pdf = self._call_unnormalized_pdf(x, name="_call_unnormalized_pdf")
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/core/basepdf.py", line 289, in _call_unnormalized_pdf
    return self._unnormalized_pdf(x)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/models/polynomials.py", line 50, in _unnormalized_pdf
    lim_low, lim_high = self.obs.limit1d
AttributeError: 'tuple' object has no attribute 'limit1d'`

Reply:
@rsilvaco Can you try again with the latest commit?

Reply:
Yes, the scaled polynomials with integral should work now (since the integral is trivially 2).

Reply:
Nop:
`Traceback (most recent call last):
  File "simpleBkgFit.py", line 156, in <module>
    minimum = minimizer.minimize(loss=nll)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 205, in minimize
    return self._hook_minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 210, in _hook_minimize
    return self._call_minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 214, in _call_minimize
    return self._minimize(loss=loss, params=params)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/minimizer_minuit.py", line 50, in _minimize
    self._check_gradients(params=params, gradients=gradients)
  File "/anaconda3/envs/zfitEnvPol/lib/python3.6/site-packages/zfit/minimizers/baseminimizer.py", line 124, in _check_gradients
    "instead of a `CompositeParameter` was created implicitly.".format(non_dependents))
ValueError: Invalid gradients for the following parameters: [<tf.Variable 'a2' shape=() dtype=float64>]The function does not depend on them. Probably a Tensorinstead of a `CompositeParameter` was created implicitly.`

Reply:
Ping @mayou36?

Reply:
Ah sorry, forgot to reply: this is "fixed".

The error is that the parameter "a2" did not had a gradient, which means the pdf was independent of it (degree number wrong implemented).

@apuignav the `self.degree` = degree - 1 seems confusing to me. What't the reason for it to be there?

An the current problem: the numeric integral of the chebyshev is not working (e.g. depends on the coefficients) -> the implementation is wrong, but I do  not see what's wrong with it. If you find time (anyone) to have a look at the implementation of the chebyshev pdf, feel free. It's short but I don't find any mistake with it